nixos/modules/services/networking/go-autoconfig.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.go-autoconfig;
+  configFile = pkgs.writeText "go-autoconfig.conf" (generators.toKeyValue { } cfg.settings);
+in {
+  options = {
+    services.go-autoconfig = {
+
+      enable = mkEnableOption "IMAP/SMTP autodiscover feature for mail clients";
+
+      port = mkOption {
+        type = types.port;
+        default = 1323;
+        description = "Port to listen on (use 0 for random port).";
+      };
+
+      domain = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = "Domainname used for autoconfig server.";
+        example = "autoconfig.example.org";
+      };
+
+      imap = {
+
+        server = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = "Domainname used for autoconfig server.";
+          example = "imap.example.org";
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 993;
+          description = "IMAP port your mail server is listening on.";
+        };
+
+      };
+
+      smtp = {
+
+        server = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = "Domainname used for autoconfig server.";
+          example = "smtp.example.org";
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 587;
+          description = "SMTP port your mail server is listening on.";
+        };
+
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd = {
+      services.create_ap = {
+        wantedBy = [ "multi-user.target" ];
+        description = "Create AP Service";
+        after = [ "network.target" ];
+        restartTriggers = [ configFile ];
+        serviceConfig = {
```

Comment:
can this have some hardening applied to it?

Reply:
no, really. can this be hardened? there's no reason this has to run as root with full system access to serve what we remember as being mostly static files.